int16 locationId
string name